pkg/oceancd/model/operator/install.go

```diff
@@ -0,0 +1,171 @@
+					Annotated: component_configs.Annotated{Annotations: map[string]string{}},
+				},
+				ExtraEnv:                  []corev1.EnvVar{},
+				NodeSelector:              map[string]string{},
+				Tolerations:               []corev1.Toleration{},
+				ImagePullSecrets:          []corev1.LocalObjectReference{},
+				ServiceAccountAnnotations: map[string]string{},
+			},
+		},
+		ArgoRolloutsConfig: component_configs.ArgoRolloutsConfig{
+			General: component_configs.ArgoRolloutsGeneralConfig{
+				Namespaced:         component_configs.Namespaced{Namespace: "argo-rollouts"},
+				Labeled:            component_configs.Labeled{Labels: map[string]string{}},
+				PodAnnotations:     map[string]string{},
+				PodLabels:          map[string]string{},
+				ServiceAnnotations: map[string]string{},
+				ServiceLabels:      map[string]string{},
+			},
+			Controller: component_configs.ArgoRolloutsControllerConfig{
+				Replicas:                 pointer.Int64(1),
+				NodeSelector:             map[string]string{},
+				Tolerations:              []corev1.Toleration{},
+				ExtraArgs:                map[string]string{},
+				ExtraEnv:                 []corev1.EnvVar{},
+				ImagePullSecrets:         []corev1.LocalObjectReference{},
+				ContainerSecurityContext: &corev1.SecurityContext{},
+				Probes: component_configs.Probes{
+					LivenessProbe: &corev1.Probe{
+						ProbeHandler: corev1.ProbeHandler{
+							HTTPGet: &corev1.HTTPGetAction{
+								Path: "/healthz",
+								Port: intstr.FromString("health"),
+							},
+						},
+						InitialDelaySeconds: 30,
+						PeriodSeconds:       20,
+						FailureThreshold:    3,
+						SuccessThreshold:    1,
+						TimeoutSeconds:      10,
+					},
+					ReadinessProbe: &corev1.Probe{
+						ProbeHandler: corev1.ProbeHandler{
+							HTTPGet: &corev1.HTTPGetAction{
+								Path: "/metrics",
+								Port: intstr.FromString("metrics"),
+							},
+						},
+						InitialDelaySeconds: 15,
+						PeriodSeconds:       5,
+						FailureThreshold:    3,
+						SuccessThreshold:    1,
+						TimeoutSeconds:      4,
+					},
+				},
+			},
+			Dashboard: component_configs.ArgoRolloutsDashboardConfig{
+				Enabled:                  false,
+				Replicas:                 pointer.Int64(1),
+				NodeSelector:             map[string]string{},
+				Tolerations:              []corev1.Toleration{},
+				ExtraEnv:                 []corev1.EnvVar{},
+				ImagePullSecrets:         []corev1.LocalObjectReference{},
+				ContainerSecurityContext: &corev1.SecurityContext{},
+			},
+		},
+	}
+}
+
+type OMManifestsRequest struct {
+	Namespace string        `json:"namespace"`
+	Manager   ManagerConfig `json:"manager"`
+}
+
+func NewInstallationPayload(config *InstallationConfig) OMManifestsRequest {
+	return OMManifestsRequest{Namespace: config.OceanCDConfig.Namespace, Manager: config.OceanCDConfig.ManagerConfig}
+}
+
+type OMManifestsResponse struct {
+	OM OM `json:"om"`
+}
+
+type OM struct {
```

Comment:
change to OMManifestsItemsResponse

Reply:
but this structure will be inside the `items` field

the `om` field is redundant IMHO

we’re trying to put it on crutches